packages/swap/src/routes/v2/quote.ts

```diff
@@ -225,25 +227,61 @@ export const generateQuotes = async ({
     quoteType: 'REGULAR' as const,
   };
 
-  const [quoteResult, boostedQuoteResult, dcaQuoteResult, dcaBoostedQuoteResult] =
-    await Promise.allSettled([
-      getPoolQuote(quoteArgs),
-      estimatedBoostFeeBps && getPoolQuote({ ...quoteArgs, boostFeeBps: estimatedBoostFeeBps }),
-      dcaQuoteParams &&
-        getPoolQuote({
-          ...quoteArgs,
-          swapInputAmount: dcaQuoteParams.chunkSize,
-          quoteType: 'DCA',
-        }),
-      dcaQuoteParams &&
-        estimatedBoostFeeBps &&
-        getPoolQuote({
-          ...quoteArgs,
-          boostFeeBps: estimatedBoostFeeBps,
-          swapInputAmount: dcaQuoteParams.chunkSize,
-          quoteType: 'DCA',
-        }),
-    ]);
+  const [quoteResult, boostedQuoteResult] = await Promise.allSettled([
+    getPoolQuote(quoteArgs),
+    estimatedBoostFeeBps && getPoolQuote({ ...quoteArgs, boostFeeBps: estimatedBoostFeeBps }),
+  ]);
+  const ingressFee =
+    quoteResult.status === 'fulfilled'
+      ? quoteResult.value.includedFees.find((fee) => fee.type === 'INGRESS')
+      : undefined;
+  const egressFee =
+    quoteResult.status === 'fulfilled'
+      ? quoteResult.value.includedFees.find((fee) => fee.type === 'EGRESS')
+      : undefined;
+
+  const exchangeRate = quoteResult.status === 'fulfilled' ? quoteResult.value.estimatedPrice : '0';
+  const sumIngressFeeEgressFee = new BigNumber(ingressFee?.amount ?? '0')
+    .plus(
+      egressFee
+        ? exchangeAmount({
+            amount: BigInt(egressFee.amount),
+            exchangeRate: new BigNumber(1).dividedBy(exchangeRate),
+            srcAsset: destAsset,
+            destAsset: srcAsset,
+          }) // we're exchanging from swap.destAsset to swap.srcAsset
+        : 0,
+    )
+    .toString();
+
+  const [dcaQuoteResult, dcaBoostedQuoteResult] = await Promise.allSettled([
+    dcaQuoteParams &&
+      getPoolQuote({
+        ...quoteArgs,
+        swapInputAmount:
+          dcaQuoteParams.chunkSize +
+          BigInt(
```

Comment:
hmm this is hard to understand for me.. i think we could achieve the same in the `adjustDcaQuote` without quoting an adjusted amount, right?

Reply:
as far as i understand, the problem is that at the moment we do
```
dcaQuote.egressAmount = new BigNumber(dcaQuote.egressAmount)
      .multipliedBy(dcaQuoteParams.numberOfChunks)
```
which basically includes the fees `numberOfChunks` times, right?

Reply:
but we could calculate a `chunkOutputAmount = chunkSize * exchangeRate` and then doe something like this?

```
dcaQuote.egressAmount = new BigNumber(dcaQuote.egressAmount)
      .add(chunkOutputAmount * (dcaQuoteParams.numberOfChunks - 1))
```